core/src/main/java/org/apache/gravitino/meta/AuditInfo.java

```diff
@@ -157,11 +157,20 @@ public AuditInfo merge(AuditInfo other, boolean overwrite) {
       return this;
     }
 
-    this.creator = overwrite || this.creator == null ? other.creator : creator;
-    this.createTime = overwrite || this.createTime == null ? other.createTime : createTime;
-    this.lastModifier = overwrite || this.lastModifier == null ? other.lastModifier : lastModifier;
+    this.creator =
+        (overwrite && other.creator != null) || this.creator == null ? other.creator : creator;
+    this.createTime =
+        (overwrite && other.createTime != null) || this.createTime == null
+            ? other.createTime
+            : createTime;
+    this.lastModifier =
+        (overwrite && other.lastModifier != null) || this.lastModifier == null
+            ? other.lastModifier
+            : lastModifier;
     this.lastModifiedTime =
-        overwrite || this.lastModifiedTime == null ? other.lastModifiedTime : lastModifiedTime;
+        (overwrite && other.lastModifiedTime != null) || this.lastModifiedTime == null
+            ? other.lastModifiedTime
+            : lastModifiedTime;
```

Comment:
This disobey the original meaning of "overwrite", I don't think the fix is correct and necessary.

Reply:
It is not whether the field is null or not. It is who takes the precedence. 

Beside, normally it will not happen that all the fields are null based on the current code. It only happens in your mimic case.

Anyway, this is a behavior change, and I'm not in favor of this change.

Reply:
In the real case, one audit info is from the underlying sources, another is stored in Gravitino, if the underlying one is empty/null, we will overwrite with the one stored in Gravitino. If the underlying one is not null / empty, we will always use the underlying one.

Besides, the one stored in Gravitino will always have the necessary fields, so it will not meet the case all the fields are null.